Firmware/RTK_Surveyor/Begin.ino

```diff
@@ -504,14 +529,24 @@ void beginUART2()
 //Assign UART2 interrupts to the core 0. See: https://github.com/espressif/arduino-esp32/issues/3386
 void pinUART2Task( void *pvParameters )
 {
-  serialGNSS.setRxBufferSize(settings.uartReceiveBufferSize);
-  serialGNSS.setTimeout(settings.serialTimeoutGNSS);
-  serialGNSS.begin(settings.dataPortBaud); //UART2 on pins 16/17 for SPP. The ZED-F9P will be configured to output NMEA over its UART1 at the same rate.
-
-  //Reduce threshold value above which RX FIFO full interrupt is generated
-  //Allows more time between when the UART interrupt occurs and when the FIFO buffer overruns
-  //serialGNSS.setRxFIFOFull(50); //Available in >v2.0.5
-  uart_set_rx_full_threshold(2, 50); //uart_num, threshold
+//Note: ESP32 2.0.6 does some strange auto-bauding thing here which takes 20s to complete if there is no data for it to auto-baud.
+//      That's fine for most RTK products, but causes the Ref Stn to stall for 20s. However, it doesn't stall with ESP32 2.0.2...
+//      Note to my future self: uncomment these lines to prevent the stall if/when we upgrade to ESP32 ~2.0.6.
```

Comment:
Good notes. I'd remove the personal flair. "Uncomment these lines..." as there are lots of folks who play in the codebase.

Reply:
Bonus points if you can write code that checks the ESP32 core version with compiler guards. I should have done that on line "serialGNSS.setRxFIFOFull(50); //Available in >v2.0.5"

Reply:
That's fine. Line 534 can go.